platform/linux-generic/pktio/dpdk.c

```diff
@@ -369,6 +404,12 @@ static inline int mbuf_to_pkt(pktio_entry_t *pktio_entry,
 
 		packet_set_ts(pkt_hdr, ts);
 
+		if (pkt_set_ol_rx(pktin_cfg, pkt_hdr, mbuf)) {
```

Comment:
You could create a mask from ipv4_chksum, udp_chksum and tcp_chksum, and compare it to pktin_cfg->all_bits before calling pkt_set_ol_rx() to reduce the number of if branches.